client/my-sites/site-settings/disconnect-site/index.jsx

```diff
@@ -0,0 +1,64 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import DocumentHead from 'components/data/document-head';
+import FormattedHeader from 'components/formatted-header';
+import Main from 'components/main';
+import PaginationFlow from './pagination-flow';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+
+class DisconnectSite extends Component {
+	componentDidMount() {
+		this.verifySiteIsJetpack();
+	}
+
+	componentDidUpdate() {
+		this.verifySiteIsJetpack();
+	}
+
+	verifySiteIsJetpack() {
+		if ( this.props.siteIsJetpack === false ) {
+			this.redirectToGeneral();
+		}
+	}
+
+	redirectToGeneral = () => {
```

Comment:
@tyxla, yes, this is a very valid point, which I noted down from your previous comment in the ref PR. I mentioned it in a Note within the PR description. This is The Plan and I believe it is best to be done in general, and not as part of this flow (please correct me if I'm wrong).